### PR TITLE
Sync with the recent changes in ark-algebra on pairing and testing

### PR DESCRIPTION
**Pull request by weikengchen**
## Description

This is a followup of #114. Simon Masson @simonmasson pointed out that a positive d, which in our formula should be negative, needs to be changed as well.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Updated relevant documentation in the code
- [x] Re-reviewed `Files changed` in the Github PR explorer

N/A:
- [ ] Wrote unit tests
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`

### CODE REVIEW

**Comment by weikengchen** [comment]
I decided to do a force merge.

**Comment by weikengchen** [comment]
Ok. I actually cannot do a force merge. @Pratyush can you do so? The problem is that the no_std test appears to be surprising.

**Comment by Pratyush** [comment]
Hm there's a bunch of changes here; what's the main change?

**Comment by weikengchen** [comment]
Originally it was just to fix a typo, now I think it is to merge with the refactoring...

I know why those errors show up, and I am fixing them now...

**Comment by weikengchen** [comment]
Renamed.

File: bw6_761/LICENSE-APACHE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                              Apache License
```
Comment (Pratyush) [inline bw6_761/LICENSE-APACHE:1]: This should be a soft link, no?

Comment (weikengchen) [reply bw6_761/LICENSE-APACHE:1]: Good point. Let me change that.

File: curve-constraint-tests/LICENSE-APACHE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                              Apache License
```
Comment (Pratyush) [inline curve-constraint-tests/LICENSE-APACHE:1]: Ditto

File: curve-constraint-tests/LICENSE-MIT [hunk 3]
```diff
@@ -0,0 +1,19 @@
+The MIT License (MIT)
```
Comment (Pratyush) [inline curve-constraint-tests/LICENSE-MIT:1]: Same

**Review by Pratyush — CHANGES_REQUESTED** [review]

**Comment by Pratyush** [comment]
I left a bunch of requests re: the LICENSE files

File: ed_on_bls12_381/src/curves/mod.rs [hunk 4]
```diff
@@ -70,7 +70,7 @@ impl TECurveConfig for JubjubParameters {
     /// COEFF_A = -1
     const COEFF_A: Fq = MontFp!("-1");
 
-    /// COEFF_D = (10240/10241) mod q
+    /// COEFF_D = -(10240/10241) mod q
```
Comment (Pratyush) [inline ed_on_bls12_381/src/curves/mod.rs:73]: thanks!

**Comment by Pratyush** [comment]
Thanks for handling this!

Small note, seems like the `mul_by_a` change hasn't propagated to the bandersnatch crate.

File: ed_on_bls12_381_bandersnatch/src/curves/mod.rs [hunk 5]
```diff
@@ -83,9 +83,8 @@ impl TECurveConfig for BandersnatchParameters {
 
     /// Multiplication by `a` is multiply by `-5`.
     #[inline(always)]
-    fn mul_by_a(elem: &Self::BaseField) -> Self::BaseField {
-        let t = (*elem).double().double();
-        -(t + *elem)
+    fn mul_by_a(elem: Self::BaseField) -> Self::BaseField {
+        elem.double().double() * elem;
```
Comment (Pratyush) [inline ed_on_bls12_381_bandersnatch/src/curves/mod.rs:87]:
```suggestion
        elem.double().double() * elem
```

File: curve-benches/Cargo.toml [hunk 6]
```diff
@@ -18,7 +18,7 @@ keywords = ["cryptography", "finite-fields", "elliptic-curves", "pairing"]
 categories = ["cryptography"]
 include = ["Cargo.toml", "src", "README.md", "LICENSE-APACHE", "LICENSE-MIT"]
 license = "MIT/Apache-2.0"
-edition = "2018"
```
[the remark this answers is not in the document]
Comment (weikengchen) [reply curve-benches/Cargo.toml:21]: Let me take a look.

**Comment by weikengchen** [comment]
Now depending on https://github.com/arkworks-rs/algebra/pull/465/

**Pull request merged**